apps/finance/app/src/components/Transfers.js

```diff
@@ -58,6 +89,47 @@ class Transfers extends React.PureComponent {
       ...initialState,
     })
   }
+  handleDateRangeChange = selectedDateRange => {
+    this.setState({ selectedDateRange })
+  }
+  encodeDataToCsv = (data, tokenDetails) => {
+    const csvContent = [
+      'data:text/csv;charset=utf-8,Date,Source/Recipient,Reference,Amount',
+    ]
+      .concat(
+        data.map(
+          ({
+            date,
+            numData: { amount },
+            reference,
+            isIncoming,
+            entity,
+            token,
+          }) => {
+            const { symbol, decimals } = tokenDetails[toChecksumAddress(token)]
+            const formattedAmount = formatTokenAmount(
+              amount,
+              isIncoming,
+              decimals,
+              true,
+              { rounding: 5 }
+            )
+            return `${format(
+              date,
+              'dd/MM/yy'
+            )},${entity},${reference},${`${formattedAmount} ${symbol}`}`
+          }
+        )
+      )
+      .join('\n')
+    return window.encodeURI(csvContent)
+  }
+  getCsvFilename = () => {
+    const { selectedDateRange } = this.state
+    const start = format(selectedDateRange.start, 'yyyy-MM-dd')
+    const end = format(selectedDateRange.end, 'yyyy-MM-dd')
+    return `Finance_(${start}_to_${end}).csv`
```

Comment:
Wondering we should add the proxy address to this file's name so it's unique across organizations cc @luisivan @dizzypaty

Reply:
We could also use the ENS name if available, rather than the proxy address.

Reply:
Ah yes, the org's ENS name will be hard for the Finance app to get at this point though. All we have is the Finance's app's proxy address.

Reply:
Oh true, I forgot we were in an app.